docs/api/internals.mdx

```diff
@@ -0,0 +1,546 @@
+  ```js
+  module.exports = async function () {
+    // ... global teardown code ...
+  };
+  ```
+
+### Detox lifecycle
+
+On one hand, Detox CLI can be totally agnostic about a test runner under the hood, but that would deprive us of some conveniences.
+For instance, there is a [retry mechanism][testRunner.retries], built into Detox CLI, which can schedule extra runs for failed test files.
+This requirement obliges Detox context to live longer than any test runner, and requires from the test runner to be able to report back to Detox CLI, whereas the resulting child process hierarchy can be broad and multi-tiered, e.g.:
+
+```plain text
+└── detox test ...
+    └── jest ... --maxWorkers N
+        ├── jest-worker (1)
+        ├── ...
+        └── jest-worker (N)
+```
+
+Even we run Jest directly, without Detox CLI, there's still a one-to-many relationship between its processes:
+
+```plain text
+└── jest ... --maxWorkers N
+    ├── jest-worker (1)
+    ├── ...
+    └── jest-worker (N)
+```
+
+So, if we want to be on the safe side, every process should be able to communicate with the root process, where we have the primary context of Detox, and vice versa.
+Retrying failed tests is just one of numerous needs, and there are more:
+
+* the primary context (and workers themselves, at times) needs to know how many workers are there;
+* the workers should request from the primary context to allocate a device and return it back when they are done;
+* any secondary context should be able to tell whether this is a first time it is running, or it is an _N_-th attempt already;
+
+This list can be continued and might expand even more with time, but the point is that Detox contexts will get instantiated as many times as child processes are created during the test session, and it should be something trivial to synchronize the primary and the secondary contexts to ensure a seamless experience.
+
+It is worth mentioning that Jest's main process is ill-suited for taking a device and running the tests, as its purpose is to orchestrate the entire test session and not run the tests themselves.
+This means that not all secondary contexts of Detox are "born equal" – most of them will be allocating a device for running tests, but at least one will be merely communicating with the primary context.
+
+This is exactly the reason why we call [`init()`][init] in every child process, but sometimes it is not just a simple call, but an [`init({ workerId: null })`][init] override to avoid creating a worker.
+However, the fact of initializing without a worker does not mean we can't call [`installWorker()`][installWorker] later. For example, if it turns out that Jest is running in a single process, then instead of creating two Detox contexts within the same process, we're going to reuse the existing one and just supplement it with the worker itself.
```

Comment:
All those methods are interesting. Otherwise, I wouldn't make them public. I'll consider extra elaboration on some terminology and definitions. Thanks for the feedback!